src/commands/deploy.rs

```diff
@@ -750,3 +747,15 @@ fn print_available_routes(
         }
     }
 }
+
+// Check if the token has expired.
+// If the expiration is None, assume the token has not expired
+fn has_expired(login_connection: &LoginConnection) -> bool {
+    match &login_connection.expiration {
+        Some(expiration) => match DateTime::parse_from_rfc3339(expiration) {
+            Ok(time) => Utc::now() > time,
+            Err(_) => true,
```

Comment:
So if I'm reading this correctly, if the expiration date cannot be parsed from the token string, we consider it to be expired.

Should we log an error message about the expiration date failing to be parsed? Shouldn't we inform the user that the token contains an invalid expiration date?

Reply:
currently if the token is expired (or it has failed during parsing), we initialize the login process automatically. after this log msg, this is how it would look like to user

```
$) spin deploy
Failed to parse token expiration time 'abc'
Error: premature end of input

Copy your one-time code:

2geg5Xf9

...and open the authorization page in your browser:

https://cloud.fermyon.com/device-authorization

Waiting for device authorization...

```

does this look ok?

Reply:
in a CI scenario, if this were to happen, the step will be stuck forever. should we just exit instead of restarting auth if expiration parsing has failed.

here is what we currently do when token json file is invalid:

```
$) spin deploy                    
Error: expected `,` or `}` at line 3 column 3
```

Reply:
I think we should exit when we've received invalid data.

Reply:
changed it to print following msg, and exit if parsing of expiration time fails.

```
spin deploy
Failed to parse token expiration time 'abc'. Error: premature end of input

Run `spin login` to log in again
```